### PR TITLE
Add head method to hdfilesystem

### PR DESCRIPTION
**Pull request by martindurant**

### CODE REVIEW

File: hdfs3/core.py [hunk 1]
```diff
@@ -413,6 +413,12 @@ def tail(self, path, size=None):
             f.seek(length - size)
             return f.read(size)
 
+    def head(self, path, size=None):
+        """ Return last bytes of file """
```
Comment (koverholt) [inline hdfs3/core.py]: last -> first

**Comment by koverholt** [comment]
:+1: LGTM. This is working as expected.

**Comment by koverholt** [comment]
Oops, commented on wrong PR, I get this error with head or tail, am I using them correctly?:

```
hdfs.head('/tmp/enron/edrm-enron-v2_white-s_xml.zip/merged.txt')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-15-fa4011b07f26> in <module>()
      1 with hdfs.open(filenames[0]) as f:
----> 2     hdfs.head(f)

/home/ubuntu/hdfs3/hdfs3/core.py in head(self, path, size)
    416     def head(self, path, size=None):
    417         """ Return last bytes of file """
--> 418         size = int(size) or 1024
    419         with self.open(path, 'r') as f:
    420             return f.read(size)

TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```

**Comment by koverholt** [comment]
My mistake, seems to be working when I pay attention to the arguments (e.g., size).

```
>>> hdfs.head('/tmp/enron/edrm-enron-v2_allen-p_xml.zip/merged.txt', 400)
Out[29]:
b'Date: Tue, 26 Sep 2000 09:26:00 -0700 (PDT)\r\nFrom: Phillip K Allen\r\nTo: pallen70@hotmail.com\r\nSubject: Investment Structure\r\nX-SDOC: 948896\r\nX-ZLID: zl-edrm-enron-v2-allen-p-1713.eml\r\n\r\n---------------------- Forwarded by Phillip K Allen/HOU/ECT on 09/26/2000 \r\n04:26 PM ---------------------------\r\n\r\n\r\n"George Richards" <cbpres@austin.rr.com> on 09/26/2000 01:18:45 PM\r\nPlease respond to <cbpres@au'
```

**Comment by mrocklin** [comment]
This doesn't seem like your mistake.  We're calling `int(None)`.  This looks like a bug?  It would be good to have more extensive tests on these functions.

**Comment by martindurant** [comment]
I propose we rip out all the CLI from the library, as I think we already more or less agreed. This I think is a remnant of handling that.

**Comment by mrocklin** [comment]
+1 on ripping out the CLI.  This experience is also a sign that we need to more exhaustively verify correctness of the standard API.

**Pull request merged**